crates/router/src/compatibility/stripe/payment_intents/types.rs

```diff
@@ -127,31 +134,96 @@ impl From<StripePaymentMethodDetails> for payments::PaymentMethodData {
 
 #[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
 pub struct Shipping {
-    pub address: Option<payments::AddressDetails>,
-    pub name: Option<String>,
+    pub address: AddressDetails,
+    pub name: Option<Secret<String>>,
     pub carrier: Option<String>,
-    pub phone: Option<pii::Secret<String>>,
-    pub tracking_number: Option<pii::Secret<String>>,
+    pub phone: Option<Secret<String>>,
+    pub tracking_number: Option<Secret<String>>,
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct AddressDetails {
+    pub city: Option<String>,
+    pub country: Option<api_enums::CountryAlpha2>,
+    pub line1: Option<Secret<String>>,
+    pub line2: Option<Secret<String>>,
+    pub postal_code: Option<Secret<String>>,
+    pub state: Option<Secret<String>>,
```

Comment:
If the structs here are same https://github.com/juspay/hyperswitch/pull/1654/files#diff-0265f9fa6a175bb9473f7376f51e2522f23d134181f02ab7450198a0bc2becedR20-R36  . You can use common one, don't have to create separate AddressDetails for both customer and payment_intents.

Reply:
Okay